erpnext/accounts/doctype/purchase_invoice/purchase_invoice.js

```diff
@@ -359,6 +359,7 @@ frappe.ui.form.on("Purchase Invoice", {
 	},
 
 	is_subcontracted: function(frm) {
+		frm.set_value("supplied_items", [])
```

Comment:
Don't make `supplied_items` blank here.

Instead, set blank on the server side https://github.com/frappe/erpnext/blob/master/erpnext/controllers/buying_controller.py#L163